src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -354,46 +357,46 @@ protected override INamespaceSymbol CommonCreateErrorNamespaceSymbol(INamespaceS
                 compilation = compilation.AddSyntaxTrees(syntaxTrees);
             }
 
-            Debug.Assert((object)compilation._lazyAssemblySymbol == null);
+            Debug.Assert(compilation._lazyAssemblySymbol is null);
             return compilation;
         }
 
         private CSharpCompilation(
-            string assemblyName,
+            string? assemblyName,
             CSharpCompilationOptions options,
             ImmutableArray<MetadataReference> references,
-            CSharpCompilation previousSubmission,
-            Type submissionReturnType,
-            Type hostObjectType,
+            CSharpCompilation? previousSubmission,
+            Type? submissionReturnType,
+            Type? hostObjectType,
             bool isSubmission,
-            ReferenceManager referenceManager,
+            ReferenceManager? referenceManager,
             bool reuseReferenceManager,
             SyntaxAndDeclarationManager syntaxAndDeclarations,
-            AsyncQueue<CompilationEvent> eventQueue = null)
+            AsyncQueue<CompilationEvent>? eventQueue = null)
             : this(assemblyName, options, references, previousSubmission, submissionReturnType, hostObjectType, isSubmission, referenceManager, reuseReferenceManager, syntaxAndDeclarations, SyntaxTreeCommonFeatures(syntaxAndDeclarations.ExternalSyntaxTrees), eventQueue)
         {
         }
 
         private CSharpCompilation(
-            string assemblyName,
+            string? assemblyName,
             CSharpCompilationOptions options,
             ImmutableArray<MetadataReference> references,
-            CSharpCompilation previousSubmission,
-            Type submissionReturnType,
-            Type hostObjectType,
+            CSharpCompilation? previousSubmission,
+            Type? submissionReturnType,
+            Type? hostObjectType,
             bool isSubmission,
-            ReferenceManager referenceManager,
+            ReferenceManager? referenceManager,
             bool reuseReferenceManager,
             SyntaxAndDeclarationManager syntaxAndDeclarations,
             IReadOnlyDictionary<string, string> features,
-            AsyncQueue<CompilationEvent> eventQueue = null)
+            AsyncQueue<CompilationEvent>? eventQueue = null)
```

Comment:
Interesting. Guessing this is the code which forced `eventQueue` in the base to be marked as nullable. I didn't realize that we supported this scenario.